### DISCUSSION TITLE
multi-letter key passthrough

### DISCUSSION THREAD

**Discussion by FranklinYu** [gdh1995/vimium-c#693] [Q&A] [answered]
GitHub web has some built-in shortcut like `gi` goes to “issues”, and `gp` goes to pull requests. Is it possible to map Vimium actions to `gn`? I tried that, but it eats any `g` events; even if I type `gi` GitHub doesn’t go to issues. I suspect that Vimium-C ate `g` so GitHub only sees `i`. I’d like Vimium to pass through the `gi` when it sees that `i` doesn’t match anything starting with `g`.

**Comment by gdh1995** [comment, marked as answer]
No it's impossible, because during handling `i`, the first keyboard event of `g` has been "ended", and JS extensions like Vimium C can not re-produce such a native key event perfectly.

While you may use other tricks supported by Vimium C:
1. you may `unmap gn`, `unmap gi`, and unmap all other `g`-prefixed default mappings, and then map other keys to those Vimium C commands
2. you may map `gi` to `dispatchEvent ......` when keeping `gf` default to `nextFrame`
- 1. `dispatchEvent` is a command designed to "simulate" keyboard events - extensions can not produce "native events" because of browser limits
- 2. then you may let Vimium C dispatch a sequence of key events, like:
```
map <v-de> dispatchEvent
run <v-de_g> de#key=g%2C103%2CKeyG
run <v-de_i> de#key=i%2C105%2CKeyI
run gi (de_g)(-de_g)(de_i)(-de_i)
```

Then when you press `gi`, a website will receive "simulated g and i", and it can use it or ignore it as it likes.
